includes/Admin/Cross_Origin_Isolation.php

```diff
@@ -0,0 +1,292 @@
+	}
+
+	/**
+	 * Filter the avatar tag.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $avatar      HTML for the user's avatar.
+	 * @param mixed  $id_or_email The avatar to retrieve. Accepts a user_id, Gravatar MD5 hash,
+	 *                            user email, WP_User object, WP_Post object, or WP_Comment object.
+	 * @param int    $size        Square avatar width and height in pixels to retrieve.
+	 * @param string $default     URL for the default image or a default type. Accepts '404', 'retro', 'monsterid',
+	 *                            'wavatar', 'indenticon', 'mystery', 'mm', 'mysteryman', 'blank', or
+	 *                            'gravatar_default'. Default is the value of the 'avatar_default' option, with a
+	 *                            fallback of 'mystery'.
+	 * @param string $alt         Alternative text to use in the avatar image tag. Default empty.
+	 * @param array  $args        Arguments passed to get_avatar_data(), after processing.
+	 *
+	 * @return string
+	 */
+	public function get_avatar( $avatar, $id_or_email, $size, $default, $alt, $args ) {
+		return $this->add_attribute( $avatar, 'src', $args['url'] );
+	}
+
+	/**
+	 * Do replacement to add crossorigin attribute.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $html HTML string.
+	 * @param string $attribute Attribute to check for.
+	 * @param string $url URL.
+	 *
+	 * @return string
+	 */
+	protected function add_attribute( $html, $attribute, $url ) {
+		$site_url = site_url();
+		$url      = esc_url( $url );
+
+		if ( $this->starts_with( $url, $site_url ) ) {
+			return $html;
+		}
+
+		if ( $this->starts_with( $url, '/' ) ) {
+			return $html;
+		}
+
+		return (string) str_replace( "{$attribute}='{$url}'", "crossorigin='anonymous' {$attribute}='{$url}'", $html );
+	}
+
+	/**
+	 * Does string start with.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $string       String to search.
+	 * @param string $start_string String to search with.
+	 *
+	 * @return bool
+	 */
+	private function starts_with( $string, $start_string ) {
+		$len = strlen( $start_string );
+
+		return ( substr( $string, 0, $len ) === $start_string );
+	}
+
+	/**
+	 * Is this the editor screen.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return bool
+	 */
+	protected function is_edit_screen() {
+		$screen = $this->get_current_screen();
+		if ( ! $screen ) {
+			return false;
+		}
+
+		if ( Story_Post_Type::POST_TYPE_SLUG !== $screen->post_type ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see if class is needed.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return bool
+	 */
+	protected function is_needed() {
+		if ( ! $this->experiments->is_experiment_enabled( 'videoOptimization' ) ) {
+			return false;
+		}
+
+		$user_id = get_current_user_id();
+		if ( ! $user_id ) {
+			return false;
+		}
+
+		// Backwards compatible, default values added in WP 5.5.
+		$check = get_user_meta( $user_id, User_Preferences::MEDIA_OPTIMIZATION_META_KEY, false );
```

Comment:
Why was the third argument (`$single`) changed from `true` to `false`? 🤔

Reply:
I explain in the line above. In short, if a user saved, the user meta will return as a empty string. When passed `rest_sanitize_boolean` makes it a false. So if a user as opted out and if not set, both are false. However, this is wrong. https://github.com/google/web-stories-wp/blob/main/includes/User_Preferences.php#L85